go/vt/vtgate/planbuilder/plancontext/planning_context.go

```diff
@@ -224,6 +230,89 @@ func (ctx *PlanningContext) TypeForExpr(e sqlparser.Expr) (evalengine.Type, bool
 	return t, true
 }
 
+func (ctx *PlanningContext) calculateTypeFor(e sqlparser.Expr) evalengine.Type {
+	cfg := &evalengine.Config{
+		ResolveType: func(expr sqlparser.Expr) (evalengine.Type, bool) {
+			col, isCol := expr.(*sqlparser.ColName)
+			if !isCol {
+				return evalengine.NewUnknownType(), false
+			}
+			return ctx.SemTable.TypeForExpr(col)
+		},
+		Collation:   ctx.SemTable.Collation,
+		Environment: ctx.VSchema.Environment(),
+		ResolveColumn: func(name *sqlparser.ColName) (int, error) {
+			// We don't need to resolve the column for type calculation
+			return 0, nil
+		},
+	}
+	env := evalengine.EmptyExpressionEnv(ctx.VSchema.Environment())
+
+	// We need to rewrite the aggregate functions to their corresponding types
+	// The evaluation engine compiler doesn't handle them, so we replace them with Arguments before
+	// asking the compiler for the type
+
+	// TODO: put this back in when we can calculate the aggregation types correctly
+	// expr, unknown := ctx.replaceAggrWithArg(e, cfg, env)
+	// if unknown {
+	// 	return evalengine.NewUnknownType()
+	// }
+
+	translatedExpr, err := evalengine.Translate(e, cfg)
+	if err != nil {
+		return evalengine.NewUnknownType()
+	}
+
+	typ, err := env.TypeOf(translatedExpr)
+	if err != nil {
+		return evalengine.NewUnknownType()
+	}
+	return typ
+}
+
+// replaceAggrWithArg replaces aggregate functions with Arguments in the given expression.
+// this is to prepare for sending the expression to the evalengine compiler to figure out the type
+func (ctx *PlanningContext) replaceAggrWithArg(e sqlparser.Expr, cfg *evalengine.Config, env *evalengine.ExpressionEnv) (expr sqlparser.Expr, unknown bool) {
+	expr = sqlparser.CopyOnRewrite(e, nil, func(cursor *sqlparser.CopyOnWriteCursor) {
+		agg, ok := cursor.Node().(sqlparser.AggrFunc)
+		if !ok {
```

Comment:
I see the TODO, do you want to keep this unused method before removing the TODO?

Reply:
yeah, that was my thought, but it's not important. if you think it should go, I can delete it